guides/source/active_model_basics.md

```diff
@@ -38,82 +66,368 @@ class EmailContact
 
   def deliver
     if valid?
-      # deliver email
+      # Deliver email
     end
   end
 end
 ```
 
-When including `ActiveModel::API` you get some features like:
+```irb
+irb> email_contact = EmailContact.new(name: "David", email: "david@example.com", message: "Hello World")
+
+irb> email_contact.name # Attribute Assignment
+=> "David"
 
-- model name introspection
-- conversions
-- translations
-- validations
+irb> email_contact.to_model == email_contact # Conversion
+=> true
 
-It also gives you the ability to initialize an object with a hash of attributes,
-much like any Active Record object.
+irb> email_contact.model_name.name # Naming
+=> "EmailContact"
 
-```irb
-irb> email_contact = EmailContact.new(name: 'David', email: 'david@example.com', message: 'Hello World')
-irb> email_contact.name
-=> "David"
-irb> email_contact.email
-=> "david@example.com"
-irb> email_contact.valid?
+irb> EmailContact.human_attribute_name("name") # Translation if the locale is set
+=> "Name"
+
+irb> email_contact.valid? # Validations
 => true
-irb> email_contact.persisted?
+
+irb> empty_contact = EmailContact.new
+irb> empty_contact.valid?
 => false
 ```
 
 Any class that includes `ActiveModel::API` can be used with `form_with`,
-`render` and any other Action View helper methods, just like Active Record
-objects.
+`render` and any other [Action View helper
+methods](https://api.rubyonrails.org/classes/ActionView/Helpers.html), just like
+Active Record objects.
+
+For example, `form_with` can be used to create a form for an `EmailContact`
+object as follows:
+
+```html+erb
+<%= form_with model: EmailContact.new do |form| %>
+  <%= form.text_field :name %>
+<% end %>
+```
+
+which results in the following HTML:
+
+```html
+<form action="/email_contacts" method="post">
+  <input type="text" name="email_contact[name]" id="email_contact_name">
+</form>
+```
+
+`render` can be used to render a partial with the object as a local variable:
+
+```html+erb
+<%= render "email_contact", email_contact: @email_contact %>
+```
+
+### Attributes
```

Comment:
As this isn't included in `API`, should not this be only mentioned later?

Reply:
IMO, a beginner will likely read this in order, so I'd love if the most common things appeared first in the document.